agent-inject/agent/container_sidecar_test.go

```diff
@@ -285,8 +285,9 @@ func TestContainerSidecar(t *testing.T) {
 		t.Errorf("wrong number of args, got %d, should have been %d", len(container.Args), 1)
 	}
 
-	if container.Args[0] != DefaultContainerArg {
-		t.Errorf("arg value wrong, should have been %s, got %s", DefaultContainerArg, container.Args[0])
+	arg := fmt.Sprintf(DefaultContainerArg, agent.SidecarType)
```

Comment:
Could you add different test cases under `TestContainerSidecar()` to test varying annotation sets containing 2 new annotations?
https://github.com/hashicorp/vault-k8s/blob/412e109929d5381bd77951956e7665fe8baaeaac/agent-inject/agent/annotations.go#L20-L28
https://github.com/hashicorp/vault-k8s/blob/412e109929d5381bd77951956e7665fe8baaeaac/agent-inject/agent/container_sidecar_test.go#L210-L216